proposals/10-egress-gateways.md

```diff
@@ -88,19 +88,119 @@ ingress use case.
+  - name: inject-credentials
+    type: CredentialInjector
+    phase: request-headers   # request-headers|request-body|connect|backend-request|backend-response|response-body|response-headers
+    priority: 10
+    failOpen: false
+    config:
+      secretRef:
+        name: openai-api-key
+        namespace: platform-secrets
+  - name: rate-qos
+    type: QoSController
+    phase: backend-request
+    priority: 30
+    failOpen: true
+    config:
+      requestsPerMinute: 1000
+```
+Alternatively, policies MAY be separate CRDs (e.g., `BackendTLSPolicy`, `EgressPolicy`) with `spec.targetRef: Backend`, avoiding schema growth on `Backend`.
+
+## Routing Modes
+
+### Endpoint Mode
+Client traffic flows through the egress gateway directly to an external endpoint (FQDN or IP). The gateway applies policies and routing logic before forwarding to the destination.
+
+### Parent Mode
+Client traffic flows through a local egress gateway to an upstream gateway before reaching the final endpoint. This enables gateway chaining for multi-cluster or multi-zone topologies. The local egress gateway treats the parent as a single upstream. Local retries are limited to establishing the parent connection. Request-level retries are performed by the parent. Implementations MUST prevent retry loops across gateways.
+
+## Policy Application Scopes
+
+Policies must be applicable at three levels:
+
+1. **Gateway-level**: Global rules affecting all traffic (e.g., cluster-wide CIDR restrictions, denied model lists)
+2. **Route-level**: Per-request logic via filters `HTTPRoute.rules[].filters[ExtensionRef]` (e.g., payload transforms, compliance checks)
+3. **Backend-level**: credentials, TLS, DNS, rate/QoS via `Backend.extensions` or backend-targeted policies.
+
+### Conflict Resolution
+When multiple policies influence the same request:
+- **Specificity precedence**: Route > Backend > Gateway.
+- **Same-level ties**: Implementations MUST use a deterministic tie-break (e.g., lexical name order) and surface status indicating the conflict.
+
+## AI Workload Considerations
+
+For inference and agentic workloads, the solution must support:
+
+- **[Payload Processing](../7-payload-processing.md)**: Request/response transformations (PII redaction, prompt injection detection, content filtering)
+  - note: Evaluation of payload processors occurs in the data plane; controllers reconcile objects into proxy configuration.
+- **Protocol Support**: HTTP/gRPC for inference APIs, with future consideration for MCP and A2A protocols
+- **Multi-destination Routing**: Failover between cloud providers and cross-cluster endpoints
+
+## Observability Considerations
+
+- Implementations SHOULD expose metrics tagged by `{gateway, route, backend, namespace, serviceAccount}` and surface conditions (e.g., `Accepted`, `Programmed`, `Degraded`).
+- Denials and transform failures SHOULD emit Events.
+
+## Next Steps
+
+1. Decide on Gateway resource approach (reuse vs. new EgressGateway type)
```

Comment:
we should reuse as much as we can when possible. 
this should probably be stated clearly.